### ISSUE TITLE
SPFx 1.17 - Conflicting package dependencies

### ISSUE THREAD

**Issue by StfBauer** [SharePoint/sp-dev-docs#8887]
### Target SharePoint environment

SharePoint Online

### What SharePoint development model, framework, SDK or API is this about?

💥 SharePoint Framework

### Developer environment

macOS

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [ ] 💥 Microsoft Edge
- [ ] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [ ] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### Additional environment details

- Node 16.15.1

### Describe the bug / error

![image](https://user-images.githubusercontent.com/5503835/231467786-5ad54d54-5d19-4156-ab6d-8fd4bae8a853.png)

The Microsoft dependencies conflict with each other, it is somewhat working but I guess there is a bit of a configuration messup.

### Steps to reproduce

Create a new project using 1.17.0 using No Framework

### Expected behavior

Proper dependencies.

**Comment by nick-pape** [comment]
Related: https://github.com/microsoft/rushstack/issues/3995

**Comment by nick-pape** [comment]
```
X:\test-eslint
λ npm list typescript
test-eslint@0.0.1 X:\test-eslint
├─┬ @microsoft/eslint-config-spfx@1.17.0
│ └─┬ @rushstack/eslint-config@3.2.0
│   └── typescript@5.0.4
├─┬ @microsoft/rush-stack-compiler-4.5@0.4.0
│ ├─┬ @microsoft/api-extractor@7.15.2
│ │ └── typescript@4.2.4
│ ├─┬ @rushstack/eslint-config@3.1.1
│ │ └── typescript@4.5.5 deduped invalid: ">=4.7.0" from node_modules/@microsoft/rush-stack-compiler-4.5/node_modules/@rushstack/eslint-config
│ └── typescript@4.5.5 deduped
├─┬ @rushstack/eslint-config@2.5.1
│ ├─┬ @typescript-eslint/eslint-plugin@5.6.0
│ │ └─┬ tsutils@3.21.0
│ │   └── typescript@4.5.5 deduped invalid: ">=4.7.0" from node_modules/@microsoft/rush-stack-compiler-4.5/node_modules/@rushstack/eslint-config
│ └── typescript@4.5.5 deduped
└── typescript@4.5.5
```

**Comment by nick-pape** [comment]
So some history here...

`@rushstack/eslint-config` was updated to require `typescript@>4.7.0` [around 9 months ago](https://github.com/microsoft/rushstack/pull/3565). 6 months ago, `@microsoft/rush-stack-compiler-4.5` has [its version of `@rushstack/eslint-config` bumped](https://github.com/microsoft/rushstack-legacy/commit/2d533b84bcc8f9b8fddfd8c7ffec1985e4ef8cc6). SPFx then bumped the Yeoman generator to utilize this newer version of `@microsoft/rush-stack-compiler-4.5` back in January for a different reason.

Core of the issue here is: 
```
├─┬ @microsoft/rush-stack-compiler-4.5@0.4.0
│ ├─┬ @microsoft/api-extractor@7.15.2
│ │ └── typescript@4.2.4
│ ├─┬ @rushstack/eslint-config@3.1.1
│ │ └── typescript@4.5.5 deduped invalid: ">=4.7.0" from node_modules/@microsoft/rush-stack-compiler-4.5/node_modules/@rushstack/eslint-config
```

`@microsoft/rush-stack-compiler-4.5` should be able to work with a provided `typescript@>4.5.0` but does not anymore due to this change. At this time, I think we need to downgrade its dependency on `@rushstack/eslint-config`, publish a new version, and update the generator.

--

I also noticed that `@microsoft/rush-stack-compiler-4.5` is using an older version of `@microsoft/api-extractor`, which is using `typescript@4.2.4`. We should additionally upgrade `@microsoft/rush-stack-compiler-4.5`'s dependency on `@microsoft/api-extractor` to [`7.22.2`](https://github.com/microsoft/rushstack/blame/7386ee3f2c8f8380585970bbefd8da41c80c3678/apps/api-extractor/package.json) (latest version with a `typescript@~4.5`).

**Comment by AJIXuMuK** [comment]
fixed in 1.17.2
